common/persistence/cassandra/cassandraPersistence.go

```diff
@@ -1600,49 +1602,123 @@ func (d *cassandraPersistence) ConflictResolveWorkflowExecution(request *p.Inter
 	resetWorkflow := request.ResetWorkflowSnapshot
 	newWorkflow := request.NewWorkflowSnapshot
 
-	prevRunID := request.PrevRunID
-	prevLastWriteVersion := request.PrevLastWriteVersion
-	prevState := request.PrevState
-
 	shardID := d.shardID
 
 	domainID := resetWorkflow.ExecutionInfo.DomainID
 	workflowID := resetWorkflow.ExecutionInfo.WorkflowID
 
-	executionInfo := resetWorkflow.ExecutionInfo
-	replicationState := resetWorkflow.ReplicationState
-	if newWorkflow != nil {
-		executionInfo = newWorkflow.ExecutionInfo
-		replicationState = newWorkflow.ReplicationState
+	if err := p.ValidateConflictResolveWorkflowModeState(
+		request.Mode,
+		resetWorkflow,
+		newWorkflow,
+		currentWorkflow,
+	); err != nil {
+		return err
+	}
+
+	var prevRunID string
+
+	switch request.Mode {
+	case p.ConflictResolveWorkflowModeBypassCurrent:
+		if err := d.assertNotCurrentExecution(
+			domainID,
+			workflowID,
+			resetWorkflow.ExecutionInfo.RunID); err != nil {
+			return err
+		}
+
+	case p.ConflictResolveWorkflowModeUpdateCurrent:
+		executionInfo := resetWorkflow.ExecutionInfo
+		replicationState := resetWorkflow.ReplicationState
+		if newWorkflow != nil {
+			executionInfo = newWorkflow.ExecutionInfo
+			replicationState = newWorkflow.ReplicationState
+		}
+		runID := executionInfo.RunID
+		createRequestID := executionInfo.CreateRequestID
+		state := executionInfo.State
+		closeStatus := executionInfo.CloseStatus
+		startVersion := replicationState.StartVersion
+		lastWriteVersion := replicationState.LastWriteVersion
+
+		if request.CurrentWorkflowCAS != nil {
+			prevRunID = request.CurrentWorkflowCAS.PrevRunID
+			prevLastWriteVersion := request.CurrentWorkflowCAS.PrevLastWriteVersion
+			prevState := request.CurrentWorkflowCAS.PrevState
+
+			batch.Query(templateUpdateCurrentWorkflowExecutionForNewQuery,
+				runID,
+				runID,
+				createRequestID,
+				state,
+				closeStatus,
+				startVersion,
+				lastWriteVersion,
+				lastWriteVersion,
+				state,
+				shardID,
+				rowTypeExecution,
+				domainID,
+				workflowID,
+				permanentRunID,
+				defaultVisibilityTimestamp,
+				rowTypeExecutionTaskID,
+				prevRunID,
+				prevLastWriteVersion,
+				prevState,
+			)
+		} else if currentWorkflow != nil {
+			prevRunID = currentWorkflow.ExecutionInfo.RunID
+
+			batch.Query(templateUpdateCurrentWorkflowExecutionQuery,
+				runID,
+				runID,
+				createRequestID,
+				state,
+				closeStatus,
+				startVersion,
+				lastWriteVersion,
+				lastWriteVersion,
+				state,
+				shardID,
+				rowTypeExecution,
+				domainID,
+				workflowID,
+				permanentRunID,
+				defaultVisibilityTimestamp,
+				rowTypeExecutionTaskID,
+				prevRunID,
+			)
+		} else {
```

Comment:
can this merge of the else if above and just update the prevRunID?

Reply:
yes, it can, however, it will create a if else within a if else, not sure which solution is better...